sentry_streams/sentry_streams/adapters/arroyo/reduce_delegate.py

```diff
@@ -165,26 +166,43 @@ def submit(self, message: ArroyoMessage[Union[FilteredPayload, TIn]]) -> None:
             return
         else:
             if isinstance(message.payload, RoutedValue):
+                # this is the actual raw payload
                 payload: Any = message.payload.payload
             else:
                 payload = message.payload
 
             timestamp = (
                 message.timestamp.timestamp() if message.timestamp is not None else time.time()
             )
-            msg = PyMessage(
-                payload=payload,
-                headers=[],
-                timestamp=timestamp,
-                schema=None,
-            )
+
+            if isinstance(payload, MutableSequence) and isinstance(payload[0], tuple):
+                batch = []
+                schema = None
+                for tup in payload:
+                    batch.append(tup[0])
+
+                schema = payload[0][1]
+
+                msg = PyMessage(
+                    payload=batch,
+                    headers=[],
+                    timestamp=timestamp,
+                    schema=schema,
+                )
+            else:
+                msg = PyMessage(
+                    payload=payload,
+                    headers=[],
+                    timestamp=timestamp,
+                    schema=None,
+                )
```

Comment:
Please let's write unit tests for this logic (there is already a `test_reduce_delegate` you can add onto). 
As this is a bit of a hack, a good way to keep it under control is to have this behavior well tested so that when we refactor this we will not miss corner cases.

Reply:
This test is already checking what we need here.  I made it a bit more "complex"